From 843bc9408996e53bfc173ef31c09bf4977118044 Mon Sep 17 00:00:00 2001
From: Simon Dinkin <simonx.dinkin@intel.com>
Date: Wed, 26 Oct 2016 16:57:21 +0300
Subject: [PATCH] mbo: remove reason_detail field from non-preferred channel
 report attribute

following the new changes in the mbo spec document - WFA_MBO_TechSpec_v0.0_r22:
the new spec defines to remove the option of beacon stength rssi
in the reason code field in the non-preferred channel report attribute.
according to the this removal, the reason detail field is not needed anymore
(it was used only when the reason code was beacon strength rssi).
---
 src/ap/sta_info.c            | 5 ++---
 src/ap/wnm_ap.c              | 5 ++---
 src/common/ieee802_11_defs.h | 9 ++++-----
 3 files changed, 8 insertions(+), 11 deletions(-)

diff --git a/src/ap/sta_info.c b/src/ap/sta_info.c
index 7c9855f..0f97eb5 100644
--- a/src/ap/sta_info.c
+++ b/src/ap/sta_info.c
@@ -1295,9 +1295,8 @@ void ap_sta_set_authorized(struct hostapd_data *hapd, struct sta_info *sta,
 		pos += ret;
 
 		for (info = sta->non_pref_chan; info; info = info->next) {
-			ret = os_snprintf(pos, end - pos, "non_pref_chan=%u:%u:%u:%u:",
-							  info->op_class, info->pref, info->reason_code,
-							  info->reason_detail);
+			ret = os_snprintf(pos, end - pos, "non_pref_chan=%u:%u:%u:",
+							  info->op_class, info->pref, info->reason_code);
 			count++;
 			if (os_snprintf_error(end - pos, ret))
 				break;
diff --git a/src/ap/wnm_ap.c b/src/ap/wnm_ap.c
index 823b1d1..9e59b35 100644
--- a/src/ap/wnm_ap.c
+++ b/src/ap/wnm_ap.c
@@ -447,9 +447,8 @@ static void report_notification_req(struct hostapd_data *hapd, const u8 *addr)
 	pos += ret;
 
 	for (info = sta->non_pref_chan; info; info = info->next) {
-		ret = os_snprintf(pos, end - pos, "non_pref_chan=%u:%u:%u:%u:",
-						  info->op_class, info->pref, info->reason_code,
-						  info->reason_detail);
+		ret = os_snprintf(pos, end - pos, "non_pref_chan=%u:%u:%u:",
+						  info->op_class, info->pref, info->reason_code);
 		count++;
 		if (os_snprintf_error(end - pos, ret))
 			break;
diff --git a/src/common/ieee802_11_defs.h b/src/common/ieee802_11_defs.h
index 1961eae..3241cdd 100644
--- a/src/common/ieee802_11_defs.h
+++ b/src/common/ieee802_11_defs.h
@@ -1135,13 +1135,12 @@ enum mbo_attr_id {
 /* Table 4-7: MBO AP Capability Indication Field Values */
 #define MBO_AP_CAPA_CELL_AWARE BIT(6)
 
-/* MBO v0.0_r19, 4.2.2: Non-preferred Channel Report Attribute */
-/* Table 4-10: Reason Code Field Values */
+/* MBO v0.0_r22, 4.2.2: Non-preferred Channel Report Attribute */
+/* Table 4-8: Reason Code Field Values */
 enum mbo_non_pref_chan_reason {
 	MBO_NON_PREF_CHAN_REASON_UNSPECIFIED = 0,
-	MBO_NON_PREF_CHAN_REASON_RSSI = 1,
-	MBO_NON_PREF_CHAN_REASON_EXT_INTERFERENCE = 2,
-	MBO_NON_PREF_CHAN_REASON_INT_INTERFERENCE = 3,
+	MBO_NON_PREF_CHAN_REASON_EXT_INTERFERENCE = 1,
+	MBO_NON_PREF_CHAN_REASON_INT_INTERFERENCE = 2,
 };
 
 /* MBO v0.0_r19, 4.2.3: Cellular Data Capabilities Attribute */
-- 
2.7.4

